tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/GitHubActions/GitHubCliCredentialStore.cs

```diff
@@ -0,0 +1,143 @@
+        this.account = account;
+    }
+
+    public async Task<Credentials> GetCredentials()
+    {
+        Process process = new()
+        {
+            StartInfo = GetAzureCliProcessStartInfo(),
+            EnableRaisingEvents = true
+        };
+
+        using ProcessRunner processRunner = new(process, this.processTimeout, CancellationToken.None);
+
+        string output;
+        //		try
+        //		{
+        output = await processRunner.RunAsync().ConfigureAwait(false);
+        //		}
+        //		catch (OperationCanceledException)
+        //		{
+        //			if (_isChainedCredential)
+        //			{
+        //				throw new CredentialUnavailableException(AzureCliTimeoutError);
+        //			}
+        //			else
+        //			{
+        //				throw new AuthenticationFailedException(AzureCliTimeoutError);
+        //			}
+        //		}
+        //		catch (InvalidOperationException exception)
+        //		{
+        //			bool isWinError = exception.Message.StartsWith(WinAzureCLIError, StringComparison.CurrentCultureIgnoreCase);
+        //
+        //			bool isOtherOsError = AzNotFoundPattern.IsMatch(exception.Message);
+        //
+        //			if (isWinError || isOtherOsError)
+        //			{
+        //				throw new CredentialUnavailableException(AzureCLINotInstalled);
+        //			}
+        //
+        //			bool isAADSTSError = exception.Message.Contains("AADSTS");
+        //			bool isLoginError = exception.Message.IndexOf("az login", StringComparison.OrdinalIgnoreCase) != -1 ||
+        //								exception.Message.IndexOf("az account set", StringComparison.OrdinalIgnoreCase) != -1;
+        //
+        //			if (isLoginError && !isAADSTSError)
+        //			{
+        //				throw new CredentialUnavailableException(AzNotLogIn);
+        //			}
+        //
+        //			bool isRefreshTokenFailedError = exception.Message.IndexOf(AzureCliFailedError, StringComparison.OrdinalIgnoreCase) != -1 &&
+        //											 exception.Message.IndexOf(RefreshTokeExpired, StringComparison.OrdinalIgnoreCase) != -1 ||
+        //											 exception.Message.IndexOf("CLIInternalError", StringComparison.OrdinalIgnoreCase) != -1;
+        //
+        //			if (isRefreshTokenFailedError)
+        //			{
+        //				throw new CredentialUnavailableException(InteractiveLoginRequired);
+        //			}
+        //
+        //			if (_isChainedCredential)
+        //			{
+        //				throw new CredentialUnavailableException($"{AzureCliFailedError} {Troubleshoot} {exception.Message}");
+        //			}
+        //			else
+        //			{
+        //				throw new AuthenticationFailedException($"{AzureCliFailedError} {Troubleshoot} {exception.Message}");
+        //			}
+        //		}
+
+        return new Credentials(output, AuthenticationType.Bearer);
+    }
+
+    private ProcessStartInfo GetAzureCliProcessStartInfo()
+    {
+        string environmentPath = Environment.GetEnvironmentVariable("PATH");
+
+        string command = "gh auth token";
```

Comment:
Why are we shelling out here? As opposed to just setting the token in the env or input parameter?

Reply:
This is to allow for local debugging without the user having to set an environment variable.   It's a close copy of the internals of AzureCliCredential.

Reply:
For hosted services, the token will be pulled from KV.